frontend/public/components/namespace.jsx

```diff
@@ -1044,106 +1110,104 @@ const namespaceBarDropdownDispatchToProps = (dispatch) => ({
   showStartGuide: (show) => dispatch(setFlag(FLAGS.SHOW_OPENSHIFT_START_GUIDE, show)),
 });
 
-class NamespaceBarDropdowns_ extends React.Component {
-  componentDidUpdate() {
-    const { namespace, showStartGuide } = this.props;
+const NamespaceBarDropdowns_ = (props) => {
+  const {
+    activeNamespace,
+    canCreateProject,
+    canListNS,
+    children,
+    disabled,
+    namespace,
+    onNamespaceChange,
+    setActiveNamespace,
+    showStartGuide,
+    t,
+    useProjects,
+  } = props;
+
+  React.useEffect(() => {
     if (namespace.loaded) {
       const noProjects = _.isEmpty(namespace.data);
       showStartGuide(noProjects);
     }
+  }, [namespace.data, namespace.loaded, showStartGuide]);
+
+  if (flagPending(canListNS)) {
+    return null;
   }
 
-  render() {
-    const {
-      activeNamespace,
-      onNamespaceChange,
-      setActiveNamespace,
-      canListNS,
-      canCreateProject,
-      useProjects,
-      children,
-      disabled,
-      t,
-    } = this.props;
-    if (flagPending(canListNS)) {
-      return null;
-    }
+  const { loaded, data } = namespace;
+  const model = getModel(useProjects);
+  const allNamespacesTitle =
+    model.label === 'Project' ? t('public~All Projects') : t('public~All Namespaces');
 
-    const { loaded, data } = this.props.namespace;
-    const model = getModel(useProjects);
-    const allNamespacesTitle =
-      model.label === 'Project' ? t('public~All Projects') : t('public~All Namespaces');
-    const items = {};
-    if (canListNS) {
-      items[ALL_NAMESPACES_KEY] = allNamespacesTitle;
-    }
+  let title = activeNamespace;
+  if (activeNamespace === ALL_NAMESPACES_KEY) {
+    title = allNamespacesTitle;
+  }
 
-    _.map(data, 'metadata.name')
-      .sort()
-      .forEach((name) => (items[name] = name));
+  const defaultActionItem = canCreateProject
+    ? [
+        {
+          actionTitle:
+            model.label === 'Project' ? t('public~Create Project') : t('public~Create Namespace'),
+          actionKey: CREATE_NEW_RESOURCE,
+        },
+      ]
+    : [];
```

Comment:
We can probably just remove this, see my previous comment about handling the create action in the namespace dropdown instead.
```suggestion
```

Reply:
Removed in next commit.  See above.